src/typecheck.rs

```diff
@@ -516,7 +516,26 @@ impl<'a> ActionChecker<'a> {
                 let (_, _ty) = self.infer_expr(&fake_call)?;
                 let fake_instr = self.instructions.pop().unwrap();
                 assert!(matches!(fake_instr, Instruction::CallFunction(..)));
-                self.instructions.push(Instruction::DeleteRow(*f));
+                self.instructions
+                    .push(Instruction::ChangeRow(ChangeRow::Delete, *f));
+                Ok(())
+            }
+            Action::Unextractable(f, args) => {
```

Comment:
Nits: code duplications that could potentially be avoided by extracting a common function.

The refactoring Oliver and I are working on will get rid of this type checker for actions, so this part will need to be rewritten anyway, so this is not a very big issue.